pkg/authorization/api/helpers.go

```diff
@@ -85,3 +90,89 @@ func GetPolicyBindingName(policyRefNamespace string) string {
 }
 
 var ClusterPolicyBindingName = GetPolicyBindingName("")
+
+func BuildSubjects(users, groups []string, userNameValidator, groupNameValidator validation.ValidateNameFunc) []kapi.ObjectReference {
+	subjects := []kapi.ObjectReference{}
+
+	for _, user := range users {
+		saNamespace, saName, err := serviceaccount.SplitUsername(user)
+		if err == nil {
+			subjects = append(subjects, kapi.ObjectReference{Kind: ServiceAccountKind, Namespace: saNamespace, Name: saName})
+			continue
+		}
+
+		kind := UserKind
+		if valid, _ := userNameValidator(user, false); !valid {
+			kind = SystemUserKind
+		}
+
+		subjects = append(subjects, kapi.ObjectReference{Kind: kind, Name: user})
+	}
+
+	for _, group := range groups {
+		kind := GroupKind
+		if valid, _ := groupNameValidator(group, false); !valid {
+			kind = SystemGroupKind
+		}
+
+		subjects = append(subjects, kapi.ObjectReference{Kind: kind, Name: group})
+	}
+
+	return subjects
+}
+
+// StringSubjectsFor returns users and groups for comparison against user.Info
```

Comment:
hrmm... I guess that's a little different... `defaultNamespace` may be ok with an explanation